tests/components/blebox/test_config_flow.py

```diff
@@ -0,0 +1,192 @@
+        config_flow.DOMAIN,
+        context={"source": config_entries.SOURCE_USER},
+        data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+    )
+
+    assert result["type"] == "create_entry"
+    assert result["title"] == "My gate controller"
+    assert result["data"] == {
+        config_flow.CONF_HOST: "172.2.3.4",
+        config_flow.CONF_PORT: 80,
+    }
+
+
+@pytest.fixture
+def product_class_mock():
+    """Return a mocked feature."""
+    path = "homeassistant.components.blebox.config_flow.Products"
+    patcher = patch(path, DEFAULT, blebox_uniapi.products.Products, True, True)
+    yield patcher
+
+
+async def test_flow_with_connection_failure(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(
+            side_effect=blebox_uniapi.error.ConnectionError
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_with_api_failure(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(
+            side_effect=blebox_uniapi.error.Error
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "cannot_connect"}
+
+
+async def test_flow_with_unknown_failure(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(side_effect=RuntimeError)
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "unknown"}
+
+
+async def test_flow_with_unsupported_version(hass, product_class_mock):
+    """Test that config flow works."""
+    with product_class_mock as products_class:
+        products_class.async_from_host = AsyncMock(
+            side_effect=blebox_uniapi.error.UnsupportedBoxVersion
+        )
+
+        result = await hass.config_entries.flow.async_init(
+            config_flow.DOMAIN,
+            context={"source": config_entries.SOURCE_USER},
+            data={config_flow.CONF_HOST: "172.2.3.4", config_flow.CONF_PORT: 80},
+        )
+        assert result["errors"] == {"base": "unsupported_version"}
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5462445)